server/routes/api/documents/documents.ts

```diff
@@ -394,18 +395,18 @@ router.post(
   async (ctx: APIContext<T.DocumentsInfoReq>) => {
     const { id, shareId, apiVersion } = ctx.input;
     const { user } = ctx.state;
+    const team = await getTeamFromContext(ctx);
     const { document, share, collection } = await documentLoader({
       id,
       shareId,
       user,
+      teamId: team?.id,
     });
     const isPublic = cannot(user, "read", document);
     const serializedDocument = await presentDocument(document, {
       isPublic,
     });
 
-    const team = await document.$get("team");
```

Comment:
But then the `team` is guaranteed to be the one that is associated with the share. If we just get the team from the context all that means is that it was loaded on a specific subdomain – it doesn't actually relate to the logged in user at all. Does that make sense?

Reply:
Makes sense, yes. The `share` load would fail if attempted against an incorrect `teamId`, obtained from an incorrect subdomain. The `urlId` update requires auth. So, the pair `(urlId, teamId)` used in query has an _implicit_ auth associated to it.